include/boost/gil/image_processing/threshold.hpp

```diff
@@ -420,20 +420,7 @@ void threshold_adaptive
     }
     else if (method == threshold_adaptive_method::gaussian)
     {
-        gray32f_image_t gaussian_kernel_values(kernel_size, kernel_size);
-        generate_gaussian_kernel(view(gaussian_kernel_values), 1.0);
-
-        gray32f_view_t gaussian_kernel_view = view(gaussian_kernel_values);
-        kernel_2d<float> kernel(
-            kernel_size,
-            kernel_size / 2,
-            kernel_size / 2
-        );
-
-        std::transform(gaussian_kernel_view.begin(), gaussian_kernel_view.end(), kernel.begin(),
-            [](gray32f_pixel_t pixel) -> float {return pixel.at(std::integral_constant<int, 0>{}); }
-        );
-
+        kernel_2d<float> kernel = generate_gaussian_kernel(kernel_size, 1.0);
```

Comment:
` kernel_2d<float> const kernel`?

This simple addition immediately clears potential reader's question "can a thing be modified by function call that follows?".